### PR TITLE
The number finder

### PR DESCRIPTION
**Pull request by ghost**
I still have to move PuzzleViewModel initialization out of its constructor.
ATM, when initialize PuzzleViewModel from PuzzlePage.OnAppearing(), starting text doesn't appears and button doesn't work anymore.

### CODE REVIEW

File: challengeMk2/ViewModels/PuzzleViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+            var apiResult = new TryResult
+            {
+                TryNumber = tryNumber,
+                Result = result,
+                UserTry = userTry,
+                Status = status
+            };
+
+            return apiResult;
+        }
```
Comment (jeremykairis) [inline challengeMk2/ViewModels/PuzzleViewModel.cs]: Why is this method here ? (Encapsulation !)

**Review by jeremykairis — CHANGES_REQUESTED** [review]
Please, take care of white spaces to make clean code.
Where are the animations ?

File: challengeMk2/ViewModels/PuzzleViewModel.cs (hunk 1 above)
Comment (ghost) [reply challengeMk2/ViewModels/PuzzleViewModel.cs]: Yes, the model should not have to call the api and then work on response to retrieve a valide result.
To change that, i add a puzzle service,  who will be in charge to communicate with the api and return a valide result for the view Model. 
The view Model just displays the result and choose the number for the next try before send it to Puzzle service.

Comment (ghost) [reply challengeMk2/ViewModels/PuzzleViewModel.cs]: I will create a new branch for the animations, to keep it clean ;D

**Review by jeremykairis — APPROVED** [review]

**Pull request merged**